Jenkinsfile

```diff
@@ -1,33 +1,58 @@
 pipeline {
-  agent {
-    docker {
-      image 'node:10-alpine'
-    }
+  agent { 
+    label 'linux'
   }
 
   stages {
-    stage('System info') {
-      steps {
-        sh 'node --version'
-        sh 'npm --version'
-      }
-    }
-
     stage('Install dependencies') {
       steps {
-        sh 'npm install'
+        script {
+          docker.image('node:10-alpine').inside(){
+            withEnv([
+              /* Override the npm cache directory to avoid: EACCES: permission denied, mkdir '/.npm' */
+              'npm_config_cache=npm-cache',
+              /* set home to our current directory because other bower
+              * nonsense breaks with HOME=/, e.g.:
+              * EACCES: permission denied, mkdir '/.config'
+              */
+              'HOME=.',
+              ]) {
+              sh '''node --version
+                    npm --version'''
+              sh 'npm install'
+            }
+          }
+        }
       }
     }
 
     stage('License check') {
       steps {
-        sh 'npm run license-checker'
+        script {
+          docker.image('node:10-alpine').inside(){
+            sh 'npm run license-checker'
```

Comment:
Why in the other steps you have added the `withEnv` block while here is not present?

Reply:
That's a fair point, I created this PR as a draft example of how to use ephemeral workers with docker. It happened the npm cache folder was failing and I found it was required to use the `withEnv`, I iterated a few times, and the one for the `npm run license-checker` was not required though, actually I have not spent much longer looking for the main reason though.

I got a bit skeptical about using `node.image`, I'd rather prefer using `nvm` to specify what's the environment required rather than using docker within the Jenkinsfile. The main reasons:
- easy to debug locally as the Jenkinsfile should be an orchestrator,
- running docker.image step might imply some other deeper tweaks like the one I did with the `withEnv`.

Reply:
Thank you for the insight!
For now, I'll continue to use docker, since it's one of the most common ways to use Node.js in production, while `nvm` is more used for development.
We'll see how it goes! :)